### PR TITLE
Update the limits section, remember the agent cluster

### PR DESCRIPTION
**Pull request by kvark**
This is a follow-up to #1128. We were missing the logic of remembering the current agent cluster and comparing it upon deserialization.
In the machine-specific limits section, I also removed the notion of the privacy budget. It's apparently controversial and not established. Mozilla's tentative position is that it's not useful, and WebGPU should not be based on it.

***
<a href="https://pr-preview.s3.amazonaws.com/kvark/gpuweb/pull/1147.html" title="Last updated on Oct 22, 2020, 2:06 PM UTC (8f9f227)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/1147/eb96ea6...kvark:8f9f227.html" title="Last updated on Oct 22, 2020, 2:06 PM UTC (8f9f227)">Diff</a>

### CODE REVIEW

**Review by kainino0x — APPROVED** [review]

File: spec/index.bs [hunk 1]
```diff
@@ -423,12 +420,12 @@ This includes available physical adapters, many limits on the GPU and CPU resour
 that could be used (such as the maximum texture size), and any optional hardware-specific
 capabilities that are available.
 
-This specification is designed with respect to a [privacy budget](https://github.com/bslassey/privacy-budget).
-It allows an user agent to decide how much information about the target system is exposed to
-the developer, thus mitigating the privacy exposure risks.
+User agents are not obligated to expose the real hardware limits, they are in full contol of
+how much the machine specifics are exposed. One strategy to reduce fingeprinting is binning
+all the target platforms into a few number of bins. In general, the privacy threat of exposing
```
Comment (jdashg) [inline spec/index.bs:425]: s/threat/impact/?

File: spec/index.bs [hunk 2]
```diff
@@ -1441,6 +1438,8 @@ GPUDevice includes GPUObjectBase;
         The [=device=] that this {{GPUDevice}} refers to.
 </dl>
 
+Issue: inherit from `GPUObjectBase`?
```
Comment (jdashg) [inline spec/index.bs:1441]: Is this related?

**Review by jdashg — APPROVED** [review]

File: spec/index.bs (hunk 2 above)
Comment (kvark) [reply spec/index.bs]: used to be related when I was adding the "current agent cluster" field to the object, but now it's not needed, removed

**Comment by kvark** [comment]
Thanks for all the reviews!
